plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -133,6 +137,7 @@ public IcebergTableHandle(
         this.storageProperties = ImmutableMap.copyOf(requireNonNull(storageProperties, "storageProperties is null"));
         this.recordScannedFiles = recordScannedFiles;
         this.maxScannedFileSize = requireNonNull(maxScannedFileSize, "maxScannedFileSize is null");
+        this.constraintColumns = constraintColumns;
```

Comment:
Needs a defensive copy
```suggestion
        this.constraintColumns = ImmutableSet.copyOf(constraintColumns);
```

Reply:
Thanks! Defensive copy is really safe.